share/p2p/discovery/manager.go

```diff
@@ -0,0 +1,51 @@
+package discovery
+
+import (
+	"context"
+	"errors"
+)
+
+type Manager struct {
```

Comment:
I really don't want to block on that as this PR is long overdue, but the current state is pretty messy and feels even messier since the last look I made. One of the options I see is to roll back to the last pre-refactoring version, and park cleans up in the issue. If we you see we can have some more time, then better doing it now ofc.